pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -81,6 +83,32 @@ func writeVifFile(buf []byte, pid, name string) error {
 	return nil
 }
 
+func setPodInterfaceCache(iface *v1.Interface, podInterfaceName string, uid string) error {
+	link, err := Handler.LinkByName(podInterfaceName)
+	if err != nil {
+		log.Log.Reason(err).Errorf("failed to get a link for interface: %s", podInterfaceName)
+		return err
+	}
+	// get IP address
+	addrList, err := Handler.AddrList(link, netlink.FAMILY_ALL)
+	if err != nil {
+		log.Log.Reason(err).Errorf("failed to get a link for interface: %s", podInterfaceName)
+		return err
+	}
+	// no ip assigned. ipam disabled
+	if len(addrList) == 0 {
+		return nil
+	}
+
+	err = writeToCachedFile(PodCacheInterface{Iface: iface, PodIP: addrList[0].IP.String()},
```

Comment:
For ipv6 you will get 2 addresses the global one and the link local. I'm not sure it is promised the global one is always the first to be returned.
I think it worth checking which one of the two is the global one.
Something like this -
`for _, addr := range addrList {
		if addr.IP.IsGlobalUnicastt() {
			ip = addr
		}
	}`